src/control/common/env_utils.go

```diff
@@ -0,0 +1,82 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+package common
+
+import (
+	"os"
+	"strconv"
+	"strings"
+)
+
+// ScrubEnvironment modifies the environment variables set for
+// this process and any children which inherit its environment.
+//
+// The whitelist bool determines whether the supplied list of
+// variable names is a blacklist or whitelist. If it is a blacklist,
+// then each variable in the supplied list is unset. If it is a
+// whitelist, then all variables except those in the whitelist
+// are unset.
+func ScrubEnvironment(list []string, whitelist bool) {
```

Comment:
It doesn't seem like the whitelist is actually used by any callers. Might be overkill at this time if there aren't imminent plans to use it. Plus, it looks like there isn't any shared behavior between the two cases, so it would be better split into two functions. Perhaps ScrubEnvironment (blacklist) and ScrubEnvironmentExcept (whitelist) or something like that.

Reply:
I was thinking that we'd eventually use the whitelist capability to set the ioserver environment to a set of whitelisted variables, rather than just pulling everything the way we do now.

I'm fine with splitting them up if that is clearer, though.